lib/handler/range.c

```diff
@@ -0,0 +1,124 @@
+#include <assert.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "h2o.h"
+
+
+#define CHECK_EOF()                             \
+    if (buf == buf_end) {                       \
+        *ret = -2;                              \
+        return NULL;                            \
+    }
+
+#define EXPECT_CHAR(ch)                         \
+    CHECK_EOF();                                \
+    if (*buf++ != ch) {                         \
+        *ret = -1;                              \
+        return NULL;                            \
+    }
+
+#define CHECK_OVERFLOW(range)                   \
+    if (range == SIZE_MAX) {                    \
+        *ret = -1;                              \
+        return NULL;                            \
+    }
+
+size_t *process_range(h2o_mem_pool_t *pool, h2o_iovec_t *range_value, size_t file_size, size_t* ret)
+{
+    size_t range_start = -1, range_count = 0;
+    char *buf = range_value->base, *buf_end = buf + range_value->len;
+    int good_range = 1, ows_skipped = 0;
```

Comment:
Please change the name of `ows_skipped` to something like `needs_comma` to reflect the meaning of the flag.